### PR TITLE
Make example calls inseparable from example blocks

### PR DESCRIPTION
**Pull request by bcjordan**
Adds flag to make a block un-disconnectable from its parent.

Uses flag to stick contract editor function calls to example blocks.
# Before

candisconnect-before
# After

candisconnect-examples

### CODE REVIEW

**Comment by bcjordan** [comment]
Working on core unit test covering this now.

File: blockly-core/core/ui/contract_editor/contract_editor.js [hunk 1]
```diff
@@ -408,8 +408,15 @@ Blockly.ContractEditor.prototype.registerTestResetHandler = function (testResetH
   this.testResetHandler_ = testResetHandler;
 };
 
+/**
+ * Moves a given example block to the modal blockspace.
+ * @param {Blockly.Block} exampleBlock
+ */
 Blockly.ContractEditor.prototype.addExampleBlockFromMainBlockSpace = function(exampleBlock) {
```
Comment (Bjvanminnen) [inline blockly-core/core/ui/contract_editor/contract_editor.js]: Are there any cases where this won't get called for an example block? I assume the answer is no. i.e. I assume this gets called both when adding a new example and when coming back to the level later and loading an existing example. This is why we can't get away with not persisting to xml, right?

**Comment by Bjvanminnen** [comment]
lgtm assuming the assumptions i make in my perhaps awkwardly worded comment are true.

File: blockly-core/core/ui/contract_editor/contract_editor.js (hunk 1 above)
Comment (bcjordan) [inline blockly-core/core/ui/contract_editor/contract_editor.js]: Correct!

**Pull request merged**